h/search/query.py

```diff
@@ -3,127 +3,99 @@
+            if val < 0:
+                raise ValueError
+        except (ValueError, KeyError):
+            return LIMIT_DEFAULT
+        else:
+            return val
 
-    def build(self, params):
-        """Get the resulting query object from this query builder."""
-        params = params.copy()
 
-        p_from = extract_offset(params)
-        p_size = extract_limit(params)
-        p_sort = extract_sort(params)
+class Sorter(object):
+    """
+    Sorts annotations.
 
-        filters = [f(params) for f in self.filters]
-        matchers = [m(params) for m in self.matchers]
-        aggregations = {a.key: a(params) for a in self.aggregations}
-        filters = [f for f in filters if f is not None]
-        matchers = [m for m in matchers if m is not None]
+    Sorts annotations by sort (the key to sort by)
+    and the order (the order in which to sort by).
+    """
 
-        # Remaining parameters are added as straightforward key-value matchers
-        for key, value in params.items():
-            matchers.append({"match": {key: value}})
-
-        # See https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-filtered-query.html
-        query = {
-            "bool": {
-                "filter": filters,
-                "must": matchers,
-            },
-        }
-
-        return {
-            "from": p_from,
-            "size": p_size,
-            "sort": p_sort,
-            "query": query,
-            "aggs": aggregations,
-        }
-
-
-def extract_offset(params):
-    try:
-        val = int(params.pop("offset"))
-        val = min(val, OFFSET_MAX)
-        if val < 0:
-            raise ValueError
-    except (ValueError, KeyError):
-        return 0
-    else:
-        return val
-
-
-def extract_limit(params):
-    try:
-        val = int(params.pop("limit"))
-        val = min(val, LIMIT_MAX)
-        if val < 0:
-            raise ValueError
-    except (ValueError, KeyError):
-        return LIMIT_DEFAULT
-    else:
-        return val
-
-
-def extract_sort(params):
-    sort_by = params.pop("sort", "updated")
-    # Sorting must be done on non-analyzed fields.
-    if sort_by == "user":
-        sort_by = "user_raw"
-    return [{
-        sort_by: {
-            "order": params.pop("order", "desc"),
-
-            # `unmapped_type` causes unknown fields specified as arguments to
-            # `sort` behave as if all documents contained empty values of the
-            # given type. Without this, specifying eg. `sort=foobar` throws
-            # an exception.
-            #
-            # We use the field type `boolean` to assist with migration because
-            # that exists in both ES 1 and ES 6.
-            "unmapped_type": "boolean",
-        }
-    }]
+    def __call__(self, search, params):
+        sort_by = params.pop("sort", "updated")
+        # Sorting must be done on non-analyzed fields.
+        if sort_by == "user":
+            sort_by = "user_raw"
+        return [{
```

Comment:
This function needs to be updated to follow the `__call__` signature of the other modifier classes (`__call__(self, search: elasticsearch_dsl.Search, params: MultiDict) -> elasticsearch_dsl.Search`). Currently it is still returning a list, which breaks chaining.

Reply:
^I think you reviewed this at an awkward moment in time so sorry about that. Probably shouldn't have pushed that right away. Anyways this was definitely fixed!